coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -56,10 +52,22 @@ public static OpenTelemetry initializeOpenTelemetryForCollector(
 
         var openTelemetrySdk = OpenTelemetrySdk.builder()
             .setTracerProvider(
-                SdkTracerProvider.builder().setResource(serviceResource).addSpanProcessor(spanProcessor).build()
+                SdkTracerProvider.builder()
+                    .setResource(Resource.getDefault()
+                        .toBuilder()
+                        .put(ResourceAttributes.SERVICE_NAME, serviceName)
+                        .put(ResourceAttributes.SERVICE_INSTANCE_ID, nodeName)
+                        .build())
+                    .addSpanProcessor(spanProcessor)
+                    .build()
             )
             .setMeterProvider(
-                SdkMeterProvider.builder().setResource(serviceResource).registerMetricReader(metricReader).build()
+                SdkMeterProvider.builder()
```

Comment:
🔥  The new format is much easier to read.

Give me newlines + nesting that make it easy to see what an object is composed of.